crates/pixi_command_dispatcher/src/command_dispatcher_processor/build_backend_metadata.rs

```diff
@@ -0,0 +1,116 @@
+use std::{collections::hash_map::Entry, sync::Arc};
+
+use futures::FutureExt;
+
+use super::{CommandDispatcherProcessor, PendingDeduplicatingTask, TaskResult};
+use crate::{
+    BuildBackendMetadata, BuildBackendMetadataError, CommandDispatcherError, Reporter,
+    command_dispatcher::{
+        BuildBackendMetadataId, BuildBackendMetadataTask, CommandDispatcherContext,
+    },
+};
+
+impl CommandDispatcherProcessor {
+    /// Called when a [`crate::command_dispatcher::BuildBackendMetadataTask`]
+    /// task was received.
+    pub(crate) fn on_build_backend_metadata(&mut self, task: BuildBackendMetadataTask) {
+        // Lookup the id of the source metadata to avoid deduplication.
```

Comment:
```suggestion
        // Lookup the id of the source metadata to avoid duplication.
```

Is that what you mean?